### PR TITLE
fix: convert v0 -> v1 before base32 conversion

### LINKED ISSUE #511: panic when loading from the gateway on main branch
### Steps to reproduce:

1. spin up the gateway
2. hit `http://localhost:9050/ipfs/QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG/contact` in a browser
3. gateway process logs will spit out:
```
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidCidV0Base', iroh-resolver/src/content_loader.rs:87:64
```

### PR DESCRIPTION
**Pull request by b5**
fixes #511

turns out you can't go straight from v0 -> base32 v1, which makes sense, but is irritating. Better than the alternative of silently converting into cidv1 I guess.

Anyway, fixes the gateway

### CODE REVIEW

File: iroh-resolver/src/content_loader.rs [hunk 1]
```diff
@@ -84,7 +84,11 @@ impl GatewayUrl {
     }
 
     pub fn as_url(&self, cid: &Cid) -> Url {
-        let cid_str = cid.to_string_of_base(Base::Base32Lower).unwrap();
+        let cid_str = cid
+            .into_v1()
+            .unwrap()
+            .to_string_of_base(Base::Base32Lower)
+            .unwrap();
```
Comment (dignifiedquire) [inline iroh-resolver/src/content_loader.rs:91]: are you sure this can never error?

File: iroh-resolver/src/content_loader.rs [hunk 2]
```diff
@@ -84,7 +84,11 @@ impl GatewayUrl {
     }
 
     pub fn as_url(&self, cid: &Cid) -> Url {
-        let cid_str = cid.to_string_of_base(Base::Base32Lower).unwrap();
+        let cid_str = cid
+            .into_v1()
+            .unwrap()
```
Comment (dignifiedquire) [inline iroh-resolver/src/content_loader.rs:89]: are you sure this can never error?

Comment (b5) [reply iroh-resolver/src/content_loader.rs:89]: Nope...

Comment (dignifiedquire) [reply iroh-resolver/src/content_loader.rs:89]: then though shall handle the error :)

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**